rust_toolchain/lib/dependabot/rust_toolchain/update_checker/latest_version_finder.rb

```diff
@@ -75,6 +76,13 @@ def filter_by_version_type(releases)
             # Check that the release version is in the same channel type
             next unless release_channel.channel_type == current_type
 
+            # For DatedStability channels, also check that the stability levels match
+            # e.g., "nightly-2025-08-19" should only match other "nightly-*" releases,
+            # not "stable-*" or "beta-*" releases
+            if current_type == ChannelType::DatedStability && release_channel.stability != current_channel.stability
+              next
+            end
```

Comment:
nit, Could we avoid using `rubocop:disable Metrics/PerceivedComplexity` if we ectracted this logic out into a predicate method like

```suggestion
            next if stability_levels_match?
```

Reply:
I extracted all of the predicates into their own methods. I think it's easier to follow now. Thanks for the suggestion!